### ISSUE TITLE
"Потеря" профинишировавшего спортсмена

### ISSUE THREAD

**Issue by sportog** [sportorg/pysport#444]
История произошла сегодня на установленной версии ПО, с расширенным фильтром поиска (SportOrg-v1.6.4_2023_05_17.exe).
Я запросил у пользователя файлы log, но самостоятельно не смог разобраться что могло повлечь проблему.

В 13:10 было принято решении о "закрытии" финиша, из-за некорректного учёта на старте, через 10 минут окажется что не все финишировали.
Приложение было закрыто, файл сохранен.
В 13:20 финиширует спортсмен и программа запускается заново. Происходит чтение содержимого чипа, сплит не выдается, так как генератор уже заглушен, но мы обсуждаем отметки участника.

Спустя более 2 часов уже "дома" обнаруживается что крайнего участника нет в файле... Нажималось "Сохранить" в принципе и запрашивало ли приложение при закрытии достоверно неизвестно.

Содержимое чипа спортсменки было восстановлено благодаря логу в файл /log/si20240914.log
Строчки логов из sportorg.log и sportorg_error.log прикрепляю, возможно удастся понять в чём была проблема.

[sportorg.log](https://github.com/user-attachments/files/17002675/sportorg.log)
[sportorg-errors.log](https://github.com/user-attachments/files/17002676/sportorg-errors.log)

**Comment by daxartio** [comment]
Звучит достаточно сложно, возможно повлияла фильтрация участников. Есть ли возможность повторить проблему?

**Comment by sportog** [comment]
> Звучит достаточно сложно, возможно повлияла фильтрация участников. Есть ли возможность повторить проблему?

Согласен. К сожалению, не я был за программой, просто помогал в проведении соревнований. Поэтому точная "картина" только со слов хронометриста.
Возможный виновник, "командная работа". Я забыл это уточнить. Когда серверная сторона выключается в клиентской продолжает "Командная работа" быть активной.

На следующий день соревнований при командной работе в файле 2ух-дневных соревнований программа на клиенте жутко висела пока не сделали "Сохранить как...".

В оргео почему то каждый финиш отправляется дважды, при том что на сервере указан другой адрес для отправки, так что это гарантированно делает клиент, также было ощущение что на клиенте автосохранение стало чаще указанного интервала.

И на сервере при получении финиша по командной работе не происходила автоотправка в live, приходилось каждый раз нажимать Ctrl+K. Но это я постараюсь самостоятельно выявить и подправить.
